kstreams/test_utils/test_clients.py

```diff
@@ -145,6 +150,11 @@ def partitions_for_topic(self, topic: str) -> Set:
     async def getone(
         self,
     ) -> Optional[ConsumerRecord]:  # The return type must be fixed later on
+        if self._previous_topic:
+            # Assumes previous record retrieved through getone was completed
+            self._previous_topic.task_done()
```

Comment:
I am wondering whether we can do `topic.task_done()` before `return consumer_record` instead, then we won't need `_previous_topic`. It can be argue that `task_done()` should be called only if the event was properly processed (like a commit) but I think during testing it won't make too much difference.

Reply:
We could do it, but it leaves us with risks during testing that are also there currently, namely if the consumption of an event takes a long time (asynchronously), the stream will be stopped before the event is able to be processed fully.

For example in the test `test_streams_consume_events` if we add an `await asyncio.sleep(1)` before the `save_to_db(cr.value)` call, doing `task_done()` before `return consumer_record` will then get the queue.join to return instantly, stopping the server and exiting the test before the `save_to_db(cr.value)` is made.